core/actions/data_preparation.ts

```diff
@@ -194,6 +233,46 @@ function applySessionToTableReference(
   return resolvedTableReference;
 }
 
+function applyDataPreparationDefaultsToTableReference(
+    definition: dataform.dataprep.DataPreparation,
+    tableReference: dataform.dataprep.ITableReference
+): dataform.dataprep.ITableReference {
+  // Check if defaults are defined for this data preparation
+  if (definition.configuration?.defaults) {
+    const defaults = definition.configuration?.defaults;
+    // Apply project default if needed
+    if (defaults.project && !tableReference.project) {
+      tableReference.project = defaults.project;
+    }
+    // Apply dataset default if needed
+    if (defaults.dataset && !tableReference.dataset) {
+      tableReference.dataset = defaults.dataset;
+    }
+  }
+
+  return tableReference
+}
```

Comment:
Product question: why are we allowing two places for defaults of the same properties? It seems overengineering at this stage not just to use the values from the workflow settings file